extensions/wikia/PortableInfobox/tests/PortableInfoboxParserTagControllerTest.php

```diff
@@ -178,17 +186,55 @@ public function testGetLayoutDataProvider() {
 				'message' => 'layout is an integer'
 			],
 			[
-				'layout' => [ 'layout' => [] ],
+				'layout' => [ 'layout' => [ ] ],
 				'expectedOutput' => 'pi-layout-default',
 				'text' => '<data><default>test</default></data>',
 				'message' => 'layout an empty table'
 			],
 			[
-				'layout' => [],
+				'layout' => [ ],
 				'expectedOutput' => 'pi-layout-default',
 				'text' => '<data><default>test</default></data>',
 				'message' => 'layout is not set'
 			]
 		];
 	}
+
+	/**
+	 * @dataProvider paramsDataProvider
+	 */
+	public function testParamsParsing( $expected, $params ) {
+		$text = '<data source="0"><label>0</label></data>
+    <data source="1"><label>1</label></data>
+    <data source="2"><label>2</label></data>
+    <data source="3"><label>3</label></data>';
+
+		$marker = $this->controller->renderInfobox( $text, [ ], $this->parser,
+			$this->parser->getPreprocessor()->newCustomFrame( $params ) )[ 0 ];
+		$output = $this->controller->replaceMarkers( $marker );
+
+		$result = [ ];
+		$xpath = $this->getXPath( $output );
+		$dataNodes = $xpath->query( '//aside/div[contains(@class,\'pi-data\')]' );
+		for ( $i = 0; $i < $dataNodes->length; $i++ ) {
+			$result[ $xpath->query( 'h3[contains(@class, \'pi-data-label\')]', $dataNodes->item( $i ) )->item( 0 )->nodeValue ] =
```

Comment:
The string literals here provide little information without the context of naming; A comment or extracting them to consts would be very helpful